### ISSUE TITLE
esm enabled, status shows disabled

### ISSUE THREAD

**Issue by panlinux** [CanonicalLtd/ubuntu-advantage-client#275]
On trusty, when enabling esm, status will still show it's disabled afterwards. This is because the pinning of -32768 is still there, and that is caused by an incorrect url in the auth.conf file.

In trusty, only a bare hostname entry in auth.conf will be considered for changing the pinning. That, plus how it looks up the entries in auth.conf and decides which one to use, means the file needs to have the following careful structure:
- for ESM purposes, we need one entry with a bare hostname but ending in a slash:
```
machine esm.staging.ubuntu.com/ login bearer password the-password
```
- for other services that share the same hostname but differ in the path, they must come *before*, i.e., more specific entries need to come first. So for example:
```
machine esm.staging.ubuntu.com/otherservice/ login bearer password another-password
machine esm.staging.ubuntu.com/ login bearer password a-password
```
Right now the passwords are the same, but this could change in the future, and the above rules make sure apt will select the right entry.

UA currently creates an entry like this:
```
machine esm.staging.ubuntu.com/ubuntu/ login bearer password a-password
```
That will leave the pinning at -32768 and `ua status` will claim that esm is disabled.

So to recap:
- more specific entries at the top
- for esm purposes, there needs to be one entry with just the hostname, but ending with a trailing slash, at the bottom
